Modules/arraymodule.c

```diff
@@ -292,19 +418,24 @@ u_setitem(arrayobject *ap, Py_ssize_t i, PyObject *v)
     assert(len == 1);
 
     if (i >= 0) {
-        ((wchar_t *)ap->ob_item)[i] = w;
+#if SIZEOF_WCHAR_T > 2
+        FT_ATOMIC_STORE_INT_RELAXED(((wchar_t *) items)[i], w);
```

Comment:
These are issuing warnings (see the bottom of the "Files changed" tab). I'm concerned about this approach: it mixes of sizes and data types along with issues about signedness. `wchar_t` may or may not be signed.